core/src/test/java/org/apache/druid/data/input/impl/JSONParseSpecTest.java

```diff
@@ -113,6 +113,40 @@ public void testParseRowWithConditional()
     Assert.assertEquals(expected, parsedRow);
   }
 
+  @Test
+  public void testParseRowWithNullsInMVD()
+  {
+    final JSONParseSpec parseSpec = new JSONParseSpec(
+        new TimestampSpec("timestamp", "iso", null),
+        new DimensionsSpec(DimensionsSpec.getDefaultSchemas(ImmutableList.of("foo"))),
+        new JSONPathSpec(
+            true,
+            ImmutableList.of(
+                // https://github.com/apache/druid/issues/6653 $.x.y.z where y is missing
+                new JSONPathFieldSpec(JSONPathFieldType.PATH, "foo", "$.baz.[?(@.maybe_object)].maybe_object"),
+                // https://github.com/apache/druid/issues/6653 $.x.y.z where y is null
+                new JSONPathFieldSpec(JSONPathFieldType.PATH, "nullFoo", "$.nullFoo.[?(@.value)].foo"),
+                new JSONPathFieldSpec(JSONPathFieldType.PATH, "baz", "$.baz"),
+                new JSONPathFieldSpec(JSONPathFieldType.PATH, "bar", "$.[?(@.something_else)].something_else.foo")
```

Comment:
most of these paths don't really seem meaningful for this test

I think something like `$.baz[1]` would be useful to extract a null element from the array.

if you're feeling ambitious, it might also be useful to have tests on arrays of objects and paths to extract things from them.

Reply:
Added 2 new cases